basic_autonomy_ros2/src/basic_autonomy.cpp

```diff
@@ -0,0 +1,1113 @@
+
+            std::vector<lanelet::BasicPoint2d> constrained_start_centerline(downsampled_starting_centerline.begin() + start_index_starting_centerline, downsampled_starting_centerline.begin() + end_index_starting_centerline);
+            std::vector<lanelet::BasicPoint2d> constrained_target_centerline(downsampled_target_centerline.begin() + start_index_target_centerline, downsampled_target_centerline.begin() + end_index_target_centerline);
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / constrained_start_centerline.size();
+
+            for (size_t i = 0; i < constrained_start_centerline.size(); ++i)
+            {
+                lanelet::BasicPoint2d current_position;
+                lanelet::BasicPoint2d start_lane_pt = constrained_start_centerline[i];
+                lanelet::BasicPoint2d target_lane_pt = constrained_target_centerline[i];
+                double delta = delta_step * i;
+                current_position.x() = target_lane_pt.x() * delta + (1 - delta) * start_lane_pt.x();
+                current_position.y() = target_lane_pt.y() * delta + (1 - delta) * start_lane_pt.y();
+
+                centerline_points.push_back(current_position);
+            }
+
+            // Add points from the remaining length of the target lanelet to provide sufficient distance for adding buffer
+            double dist_to_target_lane_end = lanelet::geometry::distance2d(centerline_points.back(), downsampled_target_centerline.back());
+            centerline_points.insert(centerline_points.end(), downsampled_target_centerline.begin() + end_index_target_centerline, downsampled_target_centerline.end());
+            
+            // If the additional distance from the remaining length of the target lanelet does not provide more than the required
+            // buffer_ending_downtrack, then also add points from the lanelet following the target lanelet
+            if (dist_to_target_lane_end < buffer_ending_downtrack) {
+                auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
+                if(!following_lanelets.empty()){
+                    //Arbitrarily choosing first following lanelet for buffer since points are only being used to fit spline
+                    auto following_lanelet_centerline = following_lanelets.front().centerline2d().basicLineString();
+                    centerline_points.insert(centerline_points.end(), following_lanelet_centerline.begin(), 
+                                                                                following_lanelet_centerline.end());
+                }
+            }
+
+            return centerline_points;
+        }
+
+       std::vector<std::vector<lanelet::BasicPoint2d>> resample_linestring_pair_to_same_size(std::vector<lanelet::BasicPoint2d>& line_1, std::vector<lanelet::BasicPoint2d>& line_2){
+            
+            rclcpp::Clock clock;
```

Comment:
This clock is never assigned a value. You can get the simulated time from this->now(), that should eliminate the need for creating a clock

Reply:
Updated (and tested/verified) to use std::chrono as discussed.